test/unit/dns.server.test.ts

```diff
@@ -4,125 +4,82 @@ import {
   });
 
-  afterAll(() => resetHostedZone());
-
   test('Hosted zone is created and hosted zone is returned', () => {
     expect(hostedZoneId.length).toBeGreaterThan(0);
     expect(/^[A-Z0-9]+/.test(hostedZoneId)).toBe(true);
   });
 
   test('createRecord() creates a record in existing hosted zone and returns changeId', async () => {
-    const changeId = await createRecord(
-      hostedZoneId,
-      RecordType.A,
-      'my-domain.starchart.com',
-      '192.168.0.1'
-    );
+    const changeId = await createRecord(RecordType.A, `my-domain.${rootDomain}`, '192.168.0.1');
     expect(changeId?.length).toBeGreaterThan(1);
   });
 
-  test('createRecord() throws an error when invalid hosted zone ID or name or value ', async () => {
-    await expect(
-      createRecord('WRONGHOSTEDZONEID', RecordType.A, 'sub2.starchart.com', '192.168.0.2')
-    ).rejects.toThrow();
+  test('createRecord() throws an error when name or value is provided ', async () => {
     await expect(
-      createRecord(hostedZoneId, RecordType.A, 'invalid_domain.starchart.com', '192.168.0.1')
+      createRecord(RecordType.A, `invalid_domain.${rootDomain}`, '192.168.0.1')
     ).rejects.toThrow();
+    await expect(createRecord(RecordType.A, `sub2.${rootDomain}`, '192-168-0-1')).rejects.toThrow();
     await expect(
-      createRecord(hostedZoneId, RecordType.A, 'sub2.starchart.com', '192-168-0-1')
-    ).rejects.toThrow();
-    await expect(
-      createRecord(hostedZoneId, RecordType.AAAA, 'sub2.starchart.com', '192.168.0.1')
+      createRecord(RecordType.AAAA, `sub2.${rootDomain}`, '192.168.0.1')
     ).rejects.toThrow();
   });
 
   test('upsertRecord() updates an existing record in hosted zone and returns changeId ', async () => {
-    await createRecord(hostedZoneId, RecordType.A, 'sub.starchart.com', '192.168.0.2');
+    await createRecord(RecordType.A, `sub.${rootDomain}`, '192.168.0.2');
     const changeId = await upsertRecord(
-      hostedZoneId,
       RecordType.AAAA,
-      'sub.starchart.com',
+      `sub.${rootDomain}`,
       '2001:db8:3333:4444:5555:6666:7777:8888'
     );
     expect(changeId?.length).toBeGreaterThan(1);
   });
 
-  test('upsertRecord() throws an error when non existing hosted zone ID or invalid name or value was provided', async () => {
-    await expect(
-      upsertRecord('WRONGHOSTEDZONEID', RecordType.A, 'sub.starchart.com', '192.168.0.3')
-    ).rejects.toThrow();
+  test('upsertRecord() throws an error when invalid name or value is provided', async () => {
     await expect(
-      upsertRecord(hostedZoneId, RecordType.A, 'invalid.domain.starchart.com', '192.168.0.2')
-    ).rejects.toThrow();
-    await expect(
-      upsertRecord(hostedZoneId, RecordType.A, 'sub.starchart.com', '192-168-0-2')
-    ).rejects.toThrow();
-    await expect(
-      upsertRecord(hostedZoneId, RecordType.CNAME, 'sub.starchart.com', '')
+      upsertRecord(RecordType.A, `invalid.domain.${rootDomain}`, '192.168.0.2')
```

Comment:
What makes this one invalid?

Reply:
Regex expression allows only `-`. So `invalid.domain` should be `invalid-domain`. And I think valid domain should be `domain.studentName.starchart.com`. I will update the regex.